### DISCUSSION TITLE
help

### DISCUSSION THREAD

**Discussion by zgc-zgc** [daijro/camoufox#63] [Leaks]
I'm experiencing issues with screen and font configurations. I'm unsure if there's an error in my setup as I'm unable to pass browserscans even after making adjustments. Furthermore, modifications to audio information are also failing to pass browserscans.

**Comment by daijro** [comment]
Hello, which tests are failing? Could you provide a snippet that I could reproduce the issue with?

Thanks!

**Comment by zgc-zgc** [comment]
I apologize, I made some mistakes. Here are my questions:
1--The font fingerprint on browserleaks changes every time, possibly because you're randomly adjusting the spacing.
2--Do the fonts need to exist on the computer to take effect?
3--There are still issues with screen fingerprint settings. The code and browserscan results are as shown in the image 
code:
from browserforge.fingerprints import Screen
constrains = Screen(max_width=1920, max_height=1080)
async with AsyncCamoufox(screen=constrains) as browser:
![1](https://github.com/user-attachments/assets/bbeb08b8-bee1-4070-b77c-55fa826fe000)

4--For audio fingerprint, do all three performance metrics need to be set? I only set the samplerate but it's not working.
Thank you for your help!

**Comment by zgc-zgc** [comment]
I previously developed a Chrome fingerprint browser through Chromium source code modification. Regarding audio fingerprinting, adjusting only the samplerate parameter was enough to successfully pass the detection mechanisms.

**Comment by zgc-zgc** [comment]
[fp_json.json](https://github.com/user-attachments/files/17606709/fp_json.json)
Here are the browser fingerprint configuration parameters that I acquired from the Chromium owner. I hope these parameters will be helpful for your work.

**Comment by daijro** [comment]
> 1--The font fingerprint on browserleaks changes every time, possibly because you're randomly adjusting the spacing.

This was by design until I implement a spacing solution that better reflects the font metrics of specific OS's. Although I will considering add a seed for the randomization.

> 2--Do the fonts need to exist on the computer to take effect?

Yes and no. Fonts added to the `fonts` parameter will be marked as true in the [FontFaceSet](https://developer.mozilla.org/en-US/docs/Web/API/FontFaceSet) API:

```js
document.fonts.check("16px Arial");
```

However, when the site attempts to use the font, it will instead fall back to the default font. If the site uses font metric fingerprinting, it would be able to identify that the font is not installed on your system.

> 3--There are still issues with screen fingerprint settings. The code and browserscan results are as shown in the image

That's strange, I can't seem to reproduce this. When you pass `debug=True`, does the generated `screen.width` and `screen.height` match the ones shown in BrowserScan?

> 4--For audio fingerprint, do all three performance metrics need to be set? I only set the samplerate but it's not working.

It should just work with the one property set:

```python
async def main():
    async with AsyncCamoufox(
        config={'AudioContext:sampleRate': 13232}
    ) as browser:
        page = await browser.new_page()
        await page.goto('https://audiofingerprint.openwpm.com')
        await asyncio.Future()  # Wait forever
```

![image](https://github.com/user-attachments/assets/11556482-fade-409d-a602-eaa71ecf24e6)
